### ISSUE TITLE
`no-fallthrough` false negative case

### ISSUE THREAD

**Issue by iancmyers** [eslint/eslint#270]
JSHint and JSLint will both warn on the following code, however the ESLint `no-fallthrough` rule will not:

``` js
switch (foo) {
case 1:
case 2:
case 3:
case 4:
    baz(); // JSHint says: Expected a 'break' statement before 'default'.
default:
    bar();
    break;
}
```

It's important that we capture this use-case, because the code above will execute `baz()` and `bar()`, which is most likely unexpected behavior.
